What: Updated process for proposals to projects

TheFoxAtWork opened the pull request:
Why:

* lack of clarity and specific steps in previous process introduced confusion.

This change addresses the need by:

* clarifying the process #609
* describing the issue types
* updating the proposal template
* clearly stating OWNERS of projects as requested in #487

ultrasaurus commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
-with thoughtfulness to accessibility across timezone and communication medium.
-We also encourage outreach outside of the group, when there are experts who
-might share insights (via invited presentation) or wish to get involved.
+   on a solution, bring the issue up for discussion in a Security TAG meeting.
```
I tried to make it clear below that discussion via slack/email/etc is welcome.  The more explicit this is, though, the harder it is to expect people to read the whole thing, so we should also balance concise instructions.

ultrasaurus commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
-   present the proposal in the "Planned Meeting" area of the
-   [meeting notes][https://github.com/cncf/tag-security#meeting-times].
-   Then at the meeting:
+   the problem, if so, let them know to join the planned meeting (see below)
```
again, let's keep meetings optional, can add "if they can" but the planned meeting should be noted on the issue and we should steer people to always update the issue -- things work well when everyone looks to the info in the issue as the source of truth, so it's friendly to new participation and leadership team or other project leaders and other groups can also look to the issue to see what is going on

TheFoxAtWork replied:
```suggestion
   the problem, if so, refer them to the issue which has content on discussions/meetings
```

ultrasaurus commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+   information. provide the issue link and encourage any feedback on your
+   proposed solution or activity.
+
+   B) Choose an upcoming meeting in **_either region_** where you or another
```
```suggestion
   B) Choose an upcoming meeting where you or another
```
let's not encode two regions!    We may someday split Europe/Africa from Americas

ultrasaurus commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
-      * Sponsor: takes responsibility to ensure that progress
-      is tracked and that outcomes are reported to the group, including
-      proposing to close the issue if there is not sufficient activity to
-      sustain the effort.
```
I dunno -- if no one is interested in sponsoring it, which typically happens if there's too much going on right now, do we want to encourage scoping/investment in getting the thing done?  In practice, I've rarely seen something well scoped without the collaboration of a Chair or TL

TheFoxAtWork replied:
IME - scoping is driven my the group or author with feedback from the Chair TL.  Doesn't mean it doesn't happen, and ultimately those conversations that drive scoping are usually with the chair/TL that is willing to sponsor (as sponsor they "sign off" on the scope as well?).  I've not seen where we don't sponsor due to timing, rather we sponsor but postpone the work.

ultrasaurus replied:
There are currently 21 [open proposals](https://github.com/cncf/tag-security/issues?q=is%3Aissue+is%3Aopen+label%3Aproposal) and I thought most were open due to lack of bandwidth. (If TL/Chairs weren't already working on so many things in parallel, then I figured some of those would be moving forward.)

TheFoxAtWork replied:
I think a lot of these no longer have interest.  its worth the time to sort through them and triage them accordingly.  thanks for bringing this up.

ultrasaurus commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+     label is removed, the "project" label is added and it is added to the
+     backlog.
+
+   B) **Close**: Closure may occur as a result of lack of interest (stale for 90
```
When originally written, this option was considered way earlier in the process.  At any time, the proposal may be closed by leadership team as out of scope of the group, but we agreed not to do that before it was discussed at a meeting (hence original order) but it we put scoping first then this doesn't feel right here.

TheFoxAtWork replied:
A lot of suggestions or proposals that come to the group arent fully formed.  Scoping allows us to be more definitive in what is or is not aligned with the charter.

ultrasaurus replied:
yes, agreed for something that is clearly within our charter.  If something comes to us that is not appropriate to the group, then we want to close early in the process.

ultrasaurus reviewed (commented):
Thanks for pulling this together!  I made a bunch of suggested edits, tried to annotate which I feel are required and which are just ideas.

The main issue that probably needs more of an edit is about closing the proposal and how/when that happens.

Also, minor point: next time, would be nice to avoid gratuitous changes in word-wrapping. It requires a lot of visual diffing, since that doesn't get screened out by github as whitespace diffs.

TheFoxAtWork commented on .github/ISSUE_TEMPLATE/proposal.md:
```diff
@@ -13,6 +13,17 @@ Impact: Describe the customer impact of the problem. Who will this help?  How wi
+* [ ] **I intend to be a project lead on this proposal if the community is interested in pursing this work.** This statement of intent does not preclude others from co-leading or becoming lead in my stead.
+
+Proposal to Project:
+* [ ] Asked to have issue assigned
```
```suggestion
```

TheFoxAtWork commented on .github/ISSUE_TEMPLATE/proposal.md:
```diff
@@ -13,6 +13,17 @@ Impact: Describe the customer impact of the problem. Who will this help?  How wi
+* [ ] **I intend to be a project lead on this proposal if the community is interested in pursing this work.** This statement of intent does not preclude others from co-leading or becoming lead in my stead.
+
+Proposal to Project:
+* [ ] Asked to have issue assigned
```
good catch - i thought i had removed this.

TheFoxAtWork commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+   issues](https://github.com/cncf/tag-security/issues) and determine if any
+   already exist. If an issue already exists that is marked as a suggestion but
+   you are interested in leading, comment on the issue and reach out in the
+   [#tag-security-triage](https://cloud-native.slack.com/archives/CJX8EBF1D)
```
I think it depends.  the general channel (to me) seems like the go-to place for help or posting things.  governance does as well.  triage has a fair amount of traffic but is really the right place for that notice - or governance?

TheFoxAtWork commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+   already exist. If an issue already exists that is marked as a suggestion but
+   you are interested in leading, comment on the issue and reach out in the
+   [#tag-security-triage](https://cloud-native.slack.com/archives/CJX8EBF1D)
+   channel to bring attention to it. Skip to the 3rd step.
```
```suggestion
   channel to bring attention to it. Skip to ask for collaboration.
```

lumjjb commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+   indicate their vote for or against. Members will have a week after the vote
+   begins to leave their vote. Quorum is taken to be 2/3 the number of companies
+   who have been active in the past month (via issue comment or meeting
+   attendance as recorded in the public minutes).
```
I'm a little unsure about the voting system, I think we haven't had a situation where this is needed yet, and wondering if it would be best to hold off on this until we see the need to... I think leaving it to the decision of leadership (or a unanimous decision among leadership) would be good enough.

lumjjb replied:
If we do decide to go with voting, I think that we need to have a bit more "system" around this. Voting needs to ensure visibility of the vote happening, maybe via email list. There needs to be proof of participation provided with the vote so we don't have to go around digging up history. We need to ensure that there isn't bad faith with editing meeting notes to include past participation since there is now an incentive to do so.

ashutosh-narkar replied:
I agree with Brandon that the final decision should be left with the leadership team. We could do a majority or 2/3rd vote in the leadership team. We can encourage community members to submit "non-binding" votes to show support or lack of it.

lumjjb commented on governance/roles.md:
```diff
@@ -20,60 +20,60 @@ All members are identified in the TAG [README](/README.md), with annotations
+  not fulfilling their documented responsibilities for more than 1 month. This
+  may be done through a super-majority vote of members, or if there are not
+  enough *active* members to get a super-majority of votes cast, then removal
+  may occur through a super-majority vote of the Chairs.
```
it's not quite clear here whether this is refering to our generic member list of members of a PROJECT role. i.e. removing from README or removing from issue/PR.

I don't think that there should be removal of folks from the README (but perhaps an active/inactive bit), since some that have made big contributions in the past but are no longer active should still be recognized as members.

lumjjb replied:
reading this over again, i think the Role specific clarifies this

lumjjb commented on governance/roles.md:
```diff
@@ -82,99 +82,88 @@ members who cannot attend a specific meeting time.
-technical leads.
+After a [Chair](#role-of-chairs) finishes their term, they transition into a
+role of Chair Emeritus. This allows previous Chairs to continue to chime in and
+provide valuable context and contributions to the TAG. A Chair Emeritus can
```
add link to new role description

TheFoxAtWork replied:
```suggestion
provide valuable context and contributions to the TAG. A [Chair Emeritus](#role-of-chair-emeriti) can
```

ashutosh-narkar commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
+   answer questions.
+
+1. **Vote, if required.** In some cases, there's consensus to accept a PR, and a
+   vote is not required. If there’s not consensus among the group, a formal vote
```
Nit: `not` -> `no`

TheFoxAtWork replied:
```suggestion
   vote is not required. If there’s no consensus among the group, a formal vote
```

lumjjb commented:
@TheFoxAtWork can we make the 1 added link to Chair Emeritus, and i think we can merge this in?

TheFoxAtWork commented on governance/roles.md:
```diff
@@ -82,99 +82,88 @@ members who cannot attend a specific meeting time.
-chair/TL slack and meetings). A Chair Emeritus has the same permissions/access as
-technical leads.
+After a [Chair](#role-of-chairs) finishes their term, they transition into a
+role of Chair Emeritus. This allows previous Chairs to continue to chime in and
```
```suggestion
role of [Chair Emeritus](#role-of-chair-emeriti). This allows previous Chairs to continue to chime in and
```

lumjjb commented:
strange, why is the CI failing now when previously it didnt.

TheFoxAtWork commented on governance/process.md:
```diff
@@ -1,144 +1,209 @@
-meeting attendance as recorded in the public minutes).
+   a comment of the PR or during a meeting, the person making the objection and
+   the person making the proposal will be given time to present their view at
+   the next meeting. If there are not objections, or if all concerns have been
```
```suggestion
   the next meeting. If there are no objections, or if all concerns have been
```

TheFoxAtWork commented:
Okay i think i have fixed EVERYTHING and added in the new issue on STAG reps v project leads. #774  
@lumjjb @ashutosh-narkar @anvega @achetal01 please review!!

ashutosh-narkar commented on governance/process.md:
```diff
@@ -1,144 +1,210 @@
-   the problem and have feedback on your proposed solution or activity.
+   the problem, if so, refer them to the issue which has content on
+   discussions/meetings where the issue will be introduced or where to stay
+   tuned for more information. provide the issue link and encourage any feedback
```
Nit: `provide` -> `Provide`

ashutosh-narkar commented on governance/process.md:
```diff
@@ -1,144 +1,210 @@
-   are interested in collaborating. This ensures that solutions are created
-   with multiple perspectives as well as verifies there is community interest
-   and energy to work on the proposal.
+     github issue and also mention of they are interested in collaborating. This
```
Nit: `of` -> `if`

ashutosh-narkar reviewed (commented):
Apart from the voting mechanism on the PRs the rest lgtm. Regarding the voting, imo we could consider something like binding and non-binding votes. The STAG leadership has the binding votes. I think the current approach could be difficult to keep track off.

TheFoxAtWork commented:
@ashutosh-narkar all done!  please review and merge!
